### ISSUE TITLE
onAfterChanges option (or something similar)

### ISSUE THREAD

**Issue by StefanJelner** [patrick-steele-idem/morphdom#223]
First of all thanks for this great library. I am using it for small reactive widgets in websites and it works like a charm.

In some scenarios i need to know, when morphing is completely finished. My approach now is to add `onElUpdated`, `onNodeAdded` and `onNodeDiscarded` and inside of the callbacks fire another callback which is throttled by request animation frame. This works for me, but is a little bit hacky in my opinion.

Would it be possible to provide an option - like `onAfterChanges` - which becomes invoked after everything has been done?

**Comment by pbowyer** [comment]
I've had a need today for `onAfterChanges` to reorder the output, because nodes I prevent from being removed and newly inserted nodes end up in the wrong order (I can't change the HTML structure so have to fix in JS).

Adding this to morphdom would be very useful. In the meantime, are you able to share your working but hacky code?

**Comment by StefanJelner** [comment]
Yes. Of course. What i do is to throttle all calls of `onElUpdated`, `onNodeAdded` and `onNodeDiscarded`. You can do this with the library `raf-throttle` to throttle it to the request animation frame or you can use something like `throttle-debounce` to to it by an amount of milliseconds.

Example:

```js
import rafThrottle from 'raf-throttle';

const onAfterChanges = rafThrottle(function() { console.log('onAfterChanges '); });

morphdom(
    document.getElementById('foo'),
    '<div>bar</div>',
    {
        onElUpdated: function() { onAfterChanges(); },
        onNodeAdded: function(node) { onAfterChanges(); return node; },
        onNodeDiscarded: function(node) { onAfterChanges(); return node; }
    }
);

**Comment by snewcomer** [comment]
`onAfterChanges` I like the suggestion.  I'd certainly be open for a PR.  If one doesn't come across, I'll try and get to it this month.

**Comment by leastbad** [comment]
Building on @StefanJelner's solution - while acknowledging that I'm hoping for an outcome where a callback fires after processing even if there are no changes - is it fair and reasonable to say that a final call to `onAfterChanges()` immediately before [line 753](https://github.com/patrick-steele-idem/morphdom/blob/84f28d0438f432cdef8cd869cd67ab916c029f2e/dist/morphdom.js#L753) would do the trick?

**Comment by StefanJelner** [comment]
If `onAfterChanges()` gets the changes as an argument, then it is possible to decide, whether to do something or not, based on the content of the argument. F.ex. if the argument is an empty array, then it is clear, that actually nothing changed, while if it is an array of changes (how they look like might be another discussion), then i know that something changed. And maybe in some scenarios it is useful to know what exactly changed. Just my thoughts.
